tests/v2/validation/extensions/provisioning/audits/audits.go

```diff
@@ -0,0 +1,340 @@
+			contextNames = append(contextNames, context)
+		}
+	}
+
+	for _, contextName := range contextNames {
+		dynamic, err := client.SwitchContext(contextName, kubeConfig)
+		assert.NoError(t, err)
+		resp, err := dynamic.Resource(corev1.SchemeGroupVersion.WithResource("pods")).Namespace("").List(context.TODO(), metav1.ListOptions{})
+		assert.NoError(t, err)
+		t.Logf("Switched Context to %v", contextName)
+		for _, pod := range resp.Items {
+			t.Logf("Pod %v", pod.GetName())
+		}
+	}
+}
+
+// AuditHostnameLength validates that the hostnames of the nodes in a cluster are of the correct length
+func AuditHostnameLength(t *testing.T, client *rancher.Client, clusterObject *provv1.Cluster) {
+	client, err := client.ReLogin()
+	require.NoError(t, err)
+
+	for _, mp := range clusterObject.Spec.RKEConfig.MachinePools {
+		n := wranglername.SafeConcatName(clusterObject.Name, mp.Name)
+		query, err := url.ParseQuery(fmt.Sprintf("labelSelector=%s=%s&fieldSelector=metadata.name=%s", capi.ClusterLabelName, clusterObject.Name, n))
+		require.NoError(t, err)
+
+		machineDeploymentsResp, err := client.Steve.SteveType("cluster.x-k8s.io.machinedeployment").List(query)
+		require.NoError(t, err)
+
+		assert.True(t, len(machineDeploymentsResp.Data) == 1)
+
+		md := &capi.MachineDeployment{}
+		require.NoError(t, steveV1.ConvertToK8sType(machineDeploymentsResp.Data[0].JSONResp, md))
+
+		query2, err := url.ParseQuery(fmt.Sprintf("labelSelector=%s=%s", capi.MachineDeploymentLabelName, md.Name))
+		require.NoError(t, err)
+
+		machineResp, err := client.Steve.SteveType("cluster.x-k8s.io.machine").List(query2)
+		require.NoError(t, err)
+
+		assert.True(t, len(machineResp.Data) > 0)
+
+		for i := range machineResp.Data {
+			m := capi.Machine{}
+			require.NoError(t, steveV1.ConvertToK8sType(machineResp.Data[i].JSONResp, &m))
+
+			assert.NotNil(t, m.Status.NodeRef)
+
+			dynamic, err := client.GetRancherDynamicClient()
+			require.NoError(t, err)
+
+			gv, err := schema.ParseGroupVersion(m.Spec.InfrastructureRef.APIVersion)
+			require.NoError(t, err)
+
+			gvr := schema.GroupVersionResource{
+				Group:    gv.Group,
+				Version:  gv.Version,
+				Resource: strings.ToLower(m.Spec.InfrastructureRef.Kind) + "s",
+			}
+
+			ustr, err := dynamic.Resource(gvr).Namespace(m.Namespace).Get(context.TODO(), m.Spec.InfrastructureRef.Name, metav1.GetOptions{})
+			require.NoError(t, err)
+
+			limit := 63
+			if mp.HostnameLengthLimit != 0 {
+				limit = mp.HostnameLengthLimit
+			} else if clusterObject.Spec.RKEConfig.MachinePoolDefaults.HostnameLengthLimit != 0 {
+				limit = clusterObject.Spec.RKEConfig.MachinePoolDefaults.HostnameLengthLimit
+			}
+
+			assert.True(t, len(m.Status.NodeRef.Name) <= limit)
+			if len(ustr.GetName()) < limit {
+				assert.True(t, m.Status.NodeRef.Name == ustr.GetName())
+			}
+		}
+	}
+}
+
+// AuditDeployment waits for a deployment to be ready
+func AuditDeployment(steveClient *steveV1.Client, deployment *steveV1.SteveAPIObject) error {
```

Comment:
This should probably go in extensions/workload, and call it `WaitForActiveDeployment`